lib/wit/deps/random/random.wit

```diff
@@ -0,0 +1,26 @@
+package wasi:random@0.2.0;
+/// WASI Random is a random data API.
+///
+/// It is intended to be portable at least between Unix-family platforms and
+/// Windows.
+interface random {
+    /// Return `len` cryptographically-secure random or pseudo-random bytes.
+    ///
+    /// This function must produce data at least as cryptographically secure and
+    /// fast as an adequately seeded cryptographically-secure pseudo-random
+    /// number generator (CSPRNG). It must not block, from the perspective of
+    /// the calling program, under any circumstances, including on the first
+    /// request and on requests for numbers of bytes. The returned data must
+    /// always be unpredictable.
```

Comment:
Probably not the right place, but my assumption is that the intention is that if the RNG needs more feeding before it can generate more data, it will return the empty list until it has been sufficiently prepped?

Reply:
I'm not entirely sure! This might be a good question to forward over to https://github.com/WebAssembly/wasi-random though.